staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -106,7 +106,11 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	}
 
 	if redirect := resp.Header.Get("Location"); redirect != "" {
-		resp.Header.Set("Location", t.rewriteURL(redirect, req.URL, req.Host))
+		targetURL, err := url.Parse(redirect)
+		if err != nil {
+			return nil, errors.NewBadRequest(fmt.Sprintf("error trying to parse Location header: %v", err))
```

Comment:
Perhaps NewInternal is more appropriate? The invalid `Location` header is caused by the server rather than initiated by the request.

Or we can just keep the old behavior, sending back the `redirect` as is.

Reply:
That is a good point, I went with NewInternalError because if there is ever a bug in how we set the `Location` header it will be easier to debug if there is an error in the server